src/app/repo-detail/repo-detail.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { RepoService } from '../repo.service';
+import { Repository } from '../repository';
+
+@Component({
+  selector: 'app-repo-detail',
+  templateUrl: './repo-detail.component.html',
+  styleUrls: ['./repo-detail.component.css']
+})
+export class RepoDetailComponent implements OnInit {
+  @Input() repo?: Repository;
+  name?: string;
+  repoDetails: string[][] = [];
+  
+  constructor(
+    private route: ActivatedRoute,
+    private repoService: RepoService
+  ) { }
+
+  ngOnInit(): void {
+    this.getRepo();
+  }
+
+  getRepo(): void {
+    this.route.params.subscribe((params) => {
+      this.name = params['name'];
+    });
+    if (this.name) {
+      this.repoService.getRepo(this.name)
+        .subscribe(repo => {
+          this.repo = repo
+          this.mapDetails()
+        });
+    }
```

Comment:
I don't think you can be certain that `this.name` will have been populated when the code on line 29 gets run. I would move the `if` block inside of the subscribe callback function. This will give you a nested subscription though which is frowned upon. You could get around this by using something like `switchMap` or `mergeMap` though I usually have to look at the docs to make sure I'm getting the behavior I want.

Reply:
would this work? 

```
  fetchRepo(): void {
    this.route.params
      .pipe(
        filter(params => params['name']),
        switchMap(params => {
          this.name = params['name'];
          return this.repoService.getRepo(params['name'])
        })
      ).subscribe(repo => {
        if (repo?.name) {
          this.repo = repo;
          this.mapDetails();
          this.repoFound = true;
        }
      })
    }
```

Reply:
also would I have to change the `ngOnInit` to `ngOnChanges` to detect changes in the route params?

Reply:
also, would I have to change the `ngOnInit` to `ngOnChanges` in order to refetch the data when the route params change?

Reply:
No need to mess with ngOnChanges. This should work, since calling `this.route.params.....subscribe(callback)` once (on init) will make sure the subscribe callback gets invoked whenever there is another value emitted.